### PR TITLE
Extend operators exercise with a unit-test-like function.

### PR DESCRIPTION
**Pull request by hageboeck**
@bernhardmgruber 

Like this maybe?

### CODE REVIEW

File: code/operators/operators.cpp [hunk 1]
```diff
@@ -7,22 +7,32 @@ class Fraction {
...
 
+
+void printAndCheck(const std::string & what, const FractionStuff::Fraction & result, const FractionStuff::Fraction & target) {
+  const bool passed = result == target;
+  std::cout << std::left << std::setw(40) << what << ": " << (passed ? "PASS" : "** FAIL **") << "    " << result << "\n";
+}
+void printAndCheck(const std::string & what, bool result, bool target) {
+  const bool passed = result == target;
+  std::cout << std::left << std::setw(40) << what << ": " << (passed ? "PASS" : "** FAIL **") << "    " << result << "\n";
+}
```
Comment (bernhardmgruber) [inline code/operators/operators.cpp]:
```suggestion
void printAndCheck(const std::string & what, const Fraction & result, const Fraction & expected) {
  const bool passed = result == expected;
  std::cout << std::left << std::setw(40) << what << ": " << (passed ? "PASS" : "** FAIL **") << "    " << expected << "\n";
}
void printAndCheck(const std::string & what, bool result, bool expected) {
  const bool passed = result == expected;
  std::cout << std::left << std::setw(40) << what << ": " << (passed ? "PASS" : "** FAIL **") << "    " << expected<< "\n";
}
```
I would suggest a different name for `target`. Furthermore, where did the namespace `FractionStuff` come from? :)
And given the redundancy here, I would almost consider using a template here.

**Review by bernhardmgruber — COMMENTED** [review]
Looks mostly good! I wondered about a but of formatting though. I used a `.clang-format` file for this exercise, but I cannot remember where I had taken it from. It looks like the Google one. Do we have one for the exercises?

File: code/operators/operators.cpp (hunk 1 above)
Comment (hageboeck) [reply code/operators/operators.cpp]: > I would suggest a different name for target. Furthermore, where did the namespace FractionStuff come from? :)

That's good, but what are you suggesting? 🙂 
It came from some playing around with ADL. Of course, that shouldn't be here! Will fix that.

> And given the redundancy here, I would almost consider using a template here.

I thought that at this point, we didn't discuss templates yet. Overloading is fine, though.

Comment (hageboeck) [reply code/operators/operators.cpp]: On formatting: I didn't use anything but vim in this one. clang-format is of course possible, but there's no official config file yet AFAIK.

Comment (bernhardmgruber) [reply code/operators/operators.cpp]: replace `target` by `expected` as I suggested in the diff.

**Comment by hageboeck** [comment]
Close in favour of https://github.com/hsf-training/cpluspluscourse/pull/100

**Pull request closed without merging**